src/backend/cdb/dispatcher/cdbdisp_query.c

```diff
@@ -278,9 +282,24 @@ cdbdisp_markNamedPortalGangsDestroyed(void)
  * Can not dispatch SET commands to busy reader gangs (allocated by cursors) directly because another
  * command is already in progress.
  * Cursors only allocate reader gangs, so primary writer and idle reader gangs can be dispatched to.
+ *
+ * isSync is true if we are QD synchronizing GUCs with QEs to bypass context
+ * constraints. See ProcessStartupPacket()
  */
 void
 CdbDispatchSetCommand(const char *strCommand, bool cancelOnError)
+{
+	return cdbdisp_dispatchSetCommandInternal(strCommand, cancelOnError, false);
+}
+
+void
+CdbDispatchSetCommandForSync(const char *strCommand)
+{
+	return cdbdisp_dispatchSetCommandInternal(strCommand, false, true);
+}
```

Comment:
put these lines after cdbdisp_dispatchSetCommandInternal and you don't need redundant declaration of this function

Reply:
Weren't there an internal discussion about some warnings due to static functions not having prototypes?

Reply:
if static function is used after defined, there is no warnings/errors